File: src/ABCStudio.ShowReels.Application/Services/ShowReelService.cs
using System;
using System.Xml.Linq;
using ABCStudio.ShowReels.Application.DTOs;
using ABCStudio.ShowReels.Core.Entities;
using ABCStudio.ShowReels.Core.Enums;
using ABCStudio.ShowReels.Core.Interfaces;

namespace ABCStudio.ShowReels.Application.Services
{
    public class ShowReelService
    {
        private readonly IShowReelRepository _showReelRepository;

        public ShowReelService(IShowReelRepository showReelRepository)
        {
            _showReelRepository = showReelRepository;
        }

        public List<ShowReelDTO> GetShowReels()
        { 
            var showReels = _showReelRepository.Get().Select(s => new ShowReelDTO
            {
                Name = s.Name,
                Description = s.Description,
                VideoClips = s.VideoClips.Select(v => new VideoClipDTO(v)).ToArray(),
                VideoDefinition = s.VideoDefinition,
                VideoStandard =s.VideoStandard
            }).ToList();

            return showReels;
        }

        public ShowReelDTO CreateShowReel(ShowReelDTO request)
        {
            var videoClips = request.VideoClips.Select(v =>
            new VideoClip(
                    v.Name,
                    v.Description,
                    v.Standard,
                    v.Definition,
                    new TimeCode(
                        v.StartTimeCode.Hours,
                        v.StartTimeCode.Minutes,
                        v.StartTimeCode.Seconds,
                        v.StartTimeCode.Frames,
                        (int)v.Standard
                        ),
                    new TimeCode(
                        v.EndTimeCode.Hours,
                        v.EndTimeCode.Minutes,
                        v.EndTimeCode.Seconds,
                        v.EndTimeCode.Frames,
                        (int)v.Standard
                        )
                )).ToList();

            var showReel = new ShowReel(
                request.Name,
                request.Description,
                request.VideoStandard,
                request.VideoDefinition,
                videoClips);

            _showReelRepository.Create(showReel);
            return new ShowReelDTO();
        }

    }
}

File: src/ABCStudio.ShowReels.Web/src/app/services/show-reel.service.ts
import { KeyValue } from '@angular/common';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { ShowReel } from '../models/show-reel';
import { TimeCode } from '../models/time-code';
import { VideoClip } from '../models/video-clip';

@Injectable({
  providedIn: 'root'
})
export class ShowReelService {

  private showReelApiUrl: string = "http://localhost:7106/api/";
  private httpOptions = {
    headers: new HttpHeaders({
      'Content-Type':  'application/json',
    })
  };
  
  constructor(private httpClient: HttpClient) { 
  }

  getVideoDefinitions(): {key: string, value: number}[] {
    return [{ key: 'SD', value: 1 }, { key: 'HD', value: 2 }];
  }

  getVideoStandards(): {key: string, value: number}[] {
    return [{ key: 'PAL', value: 25 }, { key: 'NTSC', value: 30 }];
  }
  
  get(): Observable<ShowReel[]> {
    return this.httpClient.get<ShowReel[]>(`${this.showReelApiUrl}showreel`)
      .pipe(map(showReels => showReels.map(s => { 
        var videoClips = s.videoClips.map(v => {
          var startTimeCode = new TimeCode(
            v.startTimeCode.hours, 
            v.startTimeCode.minutes, 
            v.startTimeCode.seconds, 
            v.startTimeCode.frames, 
            v.startTimeCode.framesPerSecond
          );
          var endTimeCode = new TimeCode(
            v.endTimeCode.hours, 
            v.endTimeCode.minutes, 
            v.endTimeCode.seconds, 
            v.endTimeCode.frames, 
            v.endTimeCode.framesPerSecond
          );

          return new VideoClip(
            v.name, 
            v.description, 
            v.definition,
            v.standard,
            startTimeCode,
            endTimeCode
            );
        });
        return new ShowReel(
          s.name, 
          s.description, 
          s.videoDefinition, 
          s.videoStandard, 
          videoClips);
      })));
  }

  save(showReel: ShowReel): Observable<object> {
    return this.httpClient.post(`${this.showReelApiUrl}showreel`,showReel, this.httpOptions);
  }
}
